src/webgpu/api/operation/rendering/draw.spec.ts

```diff
@@ -13,49 +13,26 @@ import {
 } from '../../../../common/util/util.js';
 import { GPUTest } from '../../../gpu_test.js';
 
-export const g = makeTestGroup(GPUTest);
+class DrawTest extends GPUTest {
+  checkTriangleDraw(
+    first_index: number | undefined,
+    count: number,
+    first_instance: number | undefined,
+    instance_count: number | undefined,
+    indexed: boolean,
+    indirect: boolean,
+    vertex_buffer_offset: number,
+    index_buffer_offset: number | undefined,
+    base_vertex: number | undefined
+  ): void {
+    // Set fallbacks when parameters are undefined in order to calculate the expected values.
+    const instanceCount = instance_count ?? 1;
+    const firstIndex = first_index ?? 0;
+    const firstInstance = first_instance ?? 0;
+    const baseVertex = base_vertex ?? 0;
+    const indexBufferOffset = index_buffer_offset ?? 0;
+    const vertexBufferOffset = vertex_buffer_offset;
```

Comment:
nit: Please use a convention other than camelCase vs snake_case to distinguish the defaulted vs non-defaulted versions. It makes it a bit hard to determine when reading whether the defaulted or non-defaulted version is being used.

If you use a named arguments pattern (without destructuring), you could do something like:

```ts
const defaulted = {
  firstIndex: opts.firstIndex ?? 0,
  count,
  firstInstance: opts.firstInstance ?? 0,
  // ...
};
```

Reply:
Thank you for suggesting it. It's better for readability!